modules/language-support/src/main/scala/playground/language/CompletionVisitor.scala

```diff
@@ -530,15 +535,55 @@ object CompletionVisitor extends SchemaVisitor[CompletionResolver] {
     fields: Vector[SchemaField[S, _]],
     make: IndexedSeq[Any] => S,
   ): CompletionResolver[S] = {
+    // Artificial schema resembling this one. Should be pretty much equivalent.
+    val schema = Schema.struct(fields)(make).addHints(hints).withId(shapeId)
+    val documentDecoder = Document.Decoder.fromSchema(schema)
+
+    val nodeEncoder = NodeEncoder.derive(schema)
+
     val compiledFields = fields.map(field => (field.mapK(this), field.instance))
 
+    /* todo: pass this outside of Hints? (visitor context?) */
+    val examples = hints
+      .get(api.Examples)
+      .foldMap(_.value)
+      .zipWithIndex
+      .map { case (example, index) =>
+        val name = example.title
+        val doc = example.documentation
+
+        val text = Formatter[InputNode]
+          .format(
+            nodeEncoder
+              .toNode(documentDecoder.decode(example.input.get).toTry.get /* todo: be graceful */ )
+              .mapK(WithSource.liftId),
+            Int.MaxValue,
+          )
+          .trim()
+          .tail
+          .init /* HACK: trim opening/closing braces */
```

Comment:
thought: a proper solution for this would probably be to type-match the encoded `InputNode` against a struct node, then format it using `FormattingVisitor`, with a new method suited specifically to structs.